great_expectations/validator/v1_validator.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from great_expectations.core.expectation_validation_result import (
+    ExpectationSuiteValidationResult,
+    ExpectationValidationResult,
+)
+from great_expectations.validator.validator import calc_validation_statistics
+
+if TYPE_CHECKING:
+    from great_expectations.core import ExpectationSuite
+    from great_expectations.core.batch_config import BatchConfig
+    from great_expectations.data_context import AbstractDataContext
+    from great_expectations.expectations.expectation import (
+        Expectation,
+        ExpectationConfiguration,
+    )
+
+
+class ResultFormat(str, Enum):
+    BOOLEAN_ONLY = "BOOLEAN_ONLY"
+    BASIC = "BASIC"
+    SUMMARY = "SUMMARY"
+    COMPLETE = "COMPLETE"
+
+
+class Validator:
+    """Validator.
+
+    Responsible for running expectations on a batch configuration.
+    """
+
+    context: AbstractDataContext
+    batch_config: BatchConfig
+    batch_asset_options: dict | None
+    result_format: ResultFormat
+
+    def __init__(
+        self,
+        context: AbstractDataContext,
+        batch_config: BatchConfig,
+        result_format: ResultFormat = ResultFormat.SUMMARY,
+        batch_asset_options: dict | None = None,
+    ) -> None:
+        self.context = context
```

Comment:
I don't think we want people mutating `context`,  `batch_config`, or `batch_asset_options`. Because of this I would prefix the instance variables with a `_`, eg `self._context = context`.

Reply:
That's fair. I originally thought all of these (except context) should function as public attributes, but thinking about it again, I'm with you. Will update.